format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -38,10 +40,20 @@
    * @param dataTypeDetectorStatusKeeper Object that keeps the state of automated data type detection process.
    * @param columnNames Column names.
    * @param rowValues Row values.
+   * @param collector Failure Collector to store failures
    */
   public static void detectDataTypeOfRowValues(Map<String, Schema> override,
                                                DataTypeDetectorStatusKeeper dataTypeDetectorStatusKeeper,
-                                               String[] columnNames, String[] rowValues) {
+                                               String[] columnNames, String[] rowValues,
+                                               @Nullable FailureCollector collector) {
+    if (rowValues.length < columnNames.length) {
+      if (collector != null) {
+        collector.addFailure(String.format(
+          "Number of values in row (%d) is less than number of values in header (%d).",
+          rowValues.length, columnNames.length), String.format("Row Values: %s", Arrays.toString(rowValues)));
```

Comment:
In this case corrective action will be to make sure row contains same number of values as columns.

Reply:
Changed.